src/meta/app/src/principal/user_privilege.rs

```diff
@@ -165,15 +179,29 @@ impl UserPrivilegeSet {
 
     /// The all privileges global which available to the table object
     pub fn available_privileges_on_table() -> Self {
-        make_bitflags!(UserPrivilegeType::{ Create | Update | Select | Insert | Delete | Drop | Alter | Grant | Ownership }).into()
+        (UserPrivilegeSet::privileges_on_table_without_ownership().privileges
+            | make_bitflags!(UserPrivilegeType::{  Ownership }))
+        .into()
     }
 
     pub fn available_privileges_on_stage() -> Self {
-        make_bitflags!(UserPrivilegeType::{  Read | Write | Ownership }).into()
+        (UserPrivilegeSet::privileges_on_stage_without_ownership().privileges
+            | make_bitflags!(UserPrivilegeType::{  Ownership }))
+        .into()
+    }
+
+    pub fn privileges_on_stage_without_ownership() -> Self {
+        make_bitflags!(UserPrivilegeType::{  Read | Write }).into()
+    }
```

Comment:
if we need seperate the cases about containing ownership or not, we can consider merge 
`privileges_on_stage_without_ownership` and `privileges_on_stage` functions with a single one like `privileges_on_stage(contain_ownership: bool)`.

Reply:
Cool.